dist/index.js

```diff
@@ -2994,48 +2994,51 @@ async function run() {
     const version = core.getInput('version');
     const ports = core.getInput('ports');
     const extra_args = core.getInput('arguments');
-    const nodeName = 'primary-node'
+    const nodeName = 'primary-node';
+    const kubeControllerManagerPodName = 'kube-controller-manager';
 
-    const kubeconfigPath=`/tmp/output/kubeconfig-${version}.yaml`;
+    const kubeconfigPath = `/tmp/output/kubeconfig-${version}.yaml`;
     console.log(`storing kubeconfig here ${kubeconfigPath}!`);
 
     const portPairs = [];
-    for(const portPair of ports.split(',')) {
-      if(portPair) {
+    for (const portPair of ports.split(',')) {
+      if (portPair) {
         portPairs.push(`-p${portPair}`);
       }
     }
 
     const extra_opts = [];
     for (const token of extra_args.split(' ')) {
       if (token) {
-        extra_opts.push(token)
+        extra_opts.push(token);
       }
     }
 
-    await exec.exec('docker', ["run","-d","--privileged",`--name=k3s-${version}`,
-      "-e",`K3S_KUBECONFIG_OUTPUT=${kubeconfigPath}`,
-      "-e","K3S_KUBECONFIG_MODE=666",
-      "-e",`K3S_NODE_NAME=${nodeName}`,
-      "-v","/tmp/output:/tmp/output",
+    await exec.exec('docker', ["run", "-d", "--privileged", `--name=k3s-${version}`,
+      "-e", `K3S_KUBECONFIG_OUTPUT=${kubeconfigPath}`,
+      "-e", "K3S_KUBECONFIG_MODE=666",
+      "-e", `K3S_NODE_NAME=${nodeName}`,
+      "-v", "/tmp/output:/tmp/output",
       "-p6443:6443",
       ...portPairs,
-      `rancher/k3s:${version}`,"server", ...extra_opts]);
+      `rancher/k3s:${version}`, "server", ...extra_opts]);
 
     core.exportVariable('KUBECONFIG', kubeconfigPath);
     core.setOutput("kubeconfig", kubeconfigPath);
 
-    while(await exec.exec(`kubectl get node/${nodeName}`, null, {ignoreReturnCode: true})) {
-      console.log('Waiting for cluster to be ready...')
+    while (await exec.exec(`kubectl get node/${nodeName}`, null, { ignoreReturnCode: true })) {
+      console.log('Waiting for cluster to be ready...');
       await wait(3000);
     }
 
+    await exec.exec(`kubectl create sa default -n default`);
```

Comment:
if the sa exists already, this will result in an error.  So far, some tests run OK and some do not - meaning the creation of the sa is inconsistent, maybe it's some kind of race condition?

Reply:
Could we do a similar check like we do for the node where we wait for the service account to be queryable/ready before finishing?  That might help determine if it's a race condition or if it is just not created for some reason.

Reply:
Thank you. I have add 
```
    while (await exec.exec(`kubectl get sa/default `, null, { ignoreReturnCode: true })) {
      console.log('Waiting for default service account to be ready...');
      await wait(3000);
    }
```
since kubectl wait is not for waiting resource creation.